openhtf/plugs/frontline/bluetooth_sniffer.py

```diff
@@ -0,0 +1,277 @@
+"""
+
+import argparse
+import contextlib
+import datetime
+import logging
+import os
+import socket
+import time
+
+from openhtf.util import files
+
+DEFAULT_AUTOMATION_PORT = 22901
+DEFAULT_HTTP_PORT = 55455
+REMOTE_BIN_PATH = (r'C:\Program Files (x86)\Frontline Test System II'
+                   r'\Frontline\Executables\Core')
+REMOTE_LOG_PATH = r'C:\ble_logs'
+VERSION = 'SNIFF00'
+
+_LOG = logging.getLogger(__name__)
+
+
+class SnifferError(Exception):
+  """Raised when a BLE sniffer command fails."""
+  def __init__(self, command, response):
+    super(SnifferError, self).__init__(
+        'Command "{}" failed with response "{}"'.format(command, response))
+    self.response = response
+
+
+class Sniffer(object):
+  """Class for controlling a Frontline bluetooth sniffer via Telnet.
+
+  Args:
+    host: Hostname or IP address of the sniffer's automation server.
+    port: TCP port fort the sniffer's automation server.
+    log_path: Path on the sniffer's automation server at which to save logs.
+  """
+  def __init__(self,
+               host='localhost',
+               port=DEFAULT_AUTOMATION_PORT,
+               log_path=REMOTE_LOG_PATH):
+    self._address = host, port
+    self._log_path = log_path
+
+    self.capture_id = None
+    self.capture_start_time = None
+    self._files = []
+    self._socket = None
+
+  def _connect(self):
+    """Open a connection to the sniffer's automation server."""
+    _LOG.info('Connecting to sniffer server at %s:%d', *self._address)
+    assert self._socket is None
+    self._socket = socket.create_connection(self._address, 10)
+
+    _LOG.info('Stopping any previously running sniffer captures')
+    self._socket.settimeout(16)
+    try:
+      self._sniffer_command('Stop FTS')
+    except (SnifferError, socket.timeout):
+      pass
+
+    self._socket.settimeout(None)
+
+  def _sniffer_command(self, command, *args):
+    """Send command to the sniffer's automation server and wait for response."""
+    full_command = ';'.join((command,) + args)
+    _LOG.debug('Sniffer command: %s', full_command)
+    self._socket.sendall(full_command.encode() + b'\n')
+
+    for line in self._socket.makefile():
+      response = line.split(';')
+      _LOG.debug('Sniffer response: %s', line.strip())
+      if command in response and 'SUCCEEDED' in response:
+        return
+      raise SnifferError(full_command, line.strip())
+
+  def _export_tab(self, tab):
+    ending = '_' + tab.lower().rsplit(':', 1)[1].replace(' ', '_') + '.csv'
+    try:
+      self._sniffer_command('Export', 'File=' + self.export_prefix + ending,
+                            'Tab=' + tab)
+    except SnifferError as e:
+      _LOG.warning('Failed to export tab %s: %s', tab, e.response)
+    else:
+      self._files.append(self.export_prefix + ending)
+
+  @property
+  def files(self):
+    """Iterator over the names of log files created by the sniffer."""
+    for path in self._files:
+      yield os.path.basename(path)
```

Comment:
i think it's a bit more readable to return a generator expression here:
`return (os.path.basename(path) for path in self._files)`

Reply:
Done.